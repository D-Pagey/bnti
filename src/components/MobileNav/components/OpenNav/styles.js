import styled from 'styled-components';
import { margin, padding } from 'polished';
import { colour } from '../../../../tokens';

export const Wrapper = styled.div`
    background-color: ${colour.blue};
    bottom: 0;
    left: 0;
    position: fixed;
    right: 0;
`;

export const List = styled.ul`
    list-style: none;
    ${margin(0)};
    ${padding(0)};
`;

export const Button = styled.button.attrs({
    type: 'button'
})``;
