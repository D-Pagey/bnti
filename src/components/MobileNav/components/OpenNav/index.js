import React from 'react';
import PropTypes from 'prop-types';
import * as S from './styles';

const OpenNav = ({ toggleOpen }) => {
    return (
        <S.Wrapper data-testid="openNav">
            <S.List>
                <li>Find Tournaments</li>
                <li>About</li>
                <li>
                    <S.Button onClick={toggleOpen}>Close Menu</S.Button>
                </li>
                <li>Sign up/Login</li>
                <li>Account</li>
            </S.List>
        </S.Wrapper>
    );
};

OpenNav.propTypes = {
    toggleOpen: PropTypes.func.isRequired
};

export default OpenNav;
