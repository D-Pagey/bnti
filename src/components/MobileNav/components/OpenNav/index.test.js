import React from 'react';
import OpenNav from '.';

describe('OpenNav component', () => {
    it('should render', () => {
        const { container } = renderWithRouter(<OpenNav />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
