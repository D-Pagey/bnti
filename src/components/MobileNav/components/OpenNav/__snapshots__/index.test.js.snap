// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenNav component should render 1`] = `
.c0 {
  background-color: #00A7CB;
  bottom: 0;
  left: 0;
  position: fixed;
  right: 0;
}

.c1 {
  list-style: none;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  padding-top: 0;
  padding-right: 0;
  padding-bottom: 0;
  padding-left: 0;
}

<div
  class="c0"
  data-testid="openNav"
>
  <ul
    class="c1"
  >
    <li>
      Find Tournaments
    </li>
    <li>
      About
    </li>
    <li>
      <button
        class=""
        type="button"
      >
        Close Menu
      </button>
    </li>
    <li>
      Sign up/Login
    </li>
    <li>
      Account
    </li>
  </ul>
</div>
`;
